pkg/prometheus/agent/operator.go

```diff
@@ -810,7 +811,8 @@ func (c *Operator) UpdateStatus(ctx context.Context, key string) error {
 		return errors.Wrap(err, "failed to get prometheus agent status")
 	}
 	p.Status = *pStatus
-	if _, err = c.mclient.MonitoringV1alpha1().PrometheusAgents(p.Namespace).UpdateStatus(ctx, p, metav1.UpdateOptions{}); err != nil {
+	paac := v1alpha1.PrometheusAgent("prometheusAgent", p.Namespace)
```

Comment:
Same here. Also IIUC the first argument should be the name of the PrometheusAgent object.

Reply:
ok. Done in other PR